engine/src/iimport.cpp

```diff
@@ -72,8 +72,65 @@ bool MCImageGetMetafileGeometry(IO_handle p_stream, uindex_t &r_width, uindex_t
 
 	if (t_success)
 	{
+        if (memcmp(t_head, "LCD\0", 4) == 0)
+        {
+            /* A drawing's header is of the form:
+             *   0: LCD\0
+             *   4: flags
+             *     bit 0: has width
+             *     bit 1: has height
+             *     bit 2: has viewport
+             *   8: scalar_count
+             *   12: width if 'has width'
+             *   16: height if 'has height'
+             *   20: viewbox if 'has viewport'.
+             *
+             * If a drawing has a width >= 0 then that is its intrinsic width
+             * Else if it has a viewport then the viewbox width is its intrinsic width
+             * Else the intrinsic width is unknown
+             *
+             * If a drawing has a height >= 0, then that is its intrinsic height
+             * Else if it has a viewport, then the viewbox height is its intrisic width
+             * Else the intrisic height is unknown
+             *
+             * Currently unknown intrinsic width/height is reported as 256.
+             */
+            uint32_t t_flags = *(uint32_t *)(t_head + 4);
+            const float *t_scalars = (const float *)(t_head + 12);
+            float t_width = -1, t_height = -1;
+            if (t_flags & 1)
+            {
+                t_width = *t_scalars++;
+            }
+            if (t_flags & 2)
+            {
+                t_height = *t_scalars++;
+            }
+            if (t_flags & 4)
+            {
+                if (t_width < 0)
+                {
+                    t_width = t_scalars[2];
+                }
+                if (t_height < 0)
+                {
+                    t_height = t_scalars[3];
+                }
```

Comment:
Might be worth a comment here to say `t_scalars[0]` and `[1]` are the viewbox origin x and y, as it took me slightly more than a moment to figure out what was going on

Reply:
Have changed to view t_scalars as an MCGRectangle* :)